pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -148,7 +149,12 @@ private Calendar getCalendar(ZoneOffset offset) {
       return calCache;
     }
 
-    calCache = new GregorianCalendar(TimeZone.getTimeZone(offset));
+    // normally we would use:
+    // calCache = new GregorianCalendar(TimeZone.getTimeZone(offset));
+    // But this seems to cause issues for some crazy offsets as returned by server for BC dates!
```

Comment:
I don't think this is a postgres specific thing java does the same ```jshell> java.time.ZoneId.of("America/Toronto").
    ...> getRules().
    ...> nextTransition(java.time.Instant.parse("0101-01-01T00:00:00Z"))
$1 ==> Transition[Gap at 1895-01-01T00:00-05:17:32 to -05:00]
```

Reply:
The problem is more that the conversion from ZoneOffset to Timezone using Timezone.get(ZoneOffset) is somehow broken. That's why I want to open a bug report.

Reply:
`05:17:32` is the geographic timezone based on latitude/longitude. Interestingly not all timezones have this rule in 1895.

Reply:
> The problem is more that the conversion from ZoneOffset to Timezone using Timezone.get(ZoneOffset) is somehow broken. 

Just guessing from my observations: To me it looks like Timezone.get() with those offsets down to second return the default Timezone! It should better throw exception when it cannot parse the ZoneOffset.

Reply:
I opened a JDK bug report about converting a ZoneOffset to TimeZone, which is buggy for offsets with seconds: https://bugs.openjdk.java.net/browse/JDK-8285844

We should maybe add a link to this issue here! @davecramer can you add the above issue to the comment above? Or do I need to open a bug report.